pallets/deitos/src/types.rs

```diff
@@ -55,15 +57,22 @@ pub enum PaymentPlanPeriods {
 #[codec(mel_bound(T: pallet::Config))]
 pub struct InfraProviderDetails<T: pallet::Config> {
     // Price per block
-    pub price_per_block: BalanceOf<T>,
+    pub price_storage_per_block: BalanceOf<T>,
+    /// Total IP Storage
+    pub total_storage: Storage,
+    // Available IP Storage
+    pub reserved_storage: Storage,
     // IP Status
     pub status: IPStatus,
+    // Track of active agreements
+    pub active_agreements: ActiveAgreements<T>,
```

Comment:
you might be wondering why i'm tracking the active_agreements here. This is a trade-off between computational  costs when reading the agreements sotrage item and actual storage cost of the bounded vector.

Reading all the active agreements per IP is something that will be usual and iterating the agreement storage is a non-go in terms of costs.

Reply:
yep that's good. will serve as a `ip => agreement` map I talked higher. But the Agreements storage should be changed to `agreement_id => agreement_terms` map to make it work. I left a comment earlier.